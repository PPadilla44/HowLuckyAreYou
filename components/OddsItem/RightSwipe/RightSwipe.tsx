import { StyleSheet, Text } from 'react-native';
import React, { FC } from 'react';
import { TouchableOpacity } from 'react-native-gesture-handler';

interface Props {
    callback: () => any;
}

const RightSwipe: FC<Props> = ({ callback }) => {
    return (
        <TouchableOpacity testID='rightSwipeBtn' onPress={callback} containerStyle={styles.btn}>
            <Text style={styles.txt}>
                Delete
            </Text>
        </TouchableOpacity>
    );
};

export default RightSwipe

const styles = StyleSheet.create({
    btn: {
        backgroundColor: 'red',
        justifyContent: 'center',
        alignItems: 'flex-end',
    },
    txt: {
        color: 'white',
        fontWeight: 'bold',
        paddingHorizontal: 24,
        fontSize: 24,
    }
})