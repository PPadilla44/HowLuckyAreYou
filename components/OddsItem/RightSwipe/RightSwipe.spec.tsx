import { cleanup, fireEvent, render, RenderAPI } from "@testing-library/react-native";
import RightSwipe from "./RightSwipe";

describe('<RightSwipe />', () => {
    

    const mockFn = jest.fn();
    let wrapper: RenderAPI;

    afterEach(cleanup)

    beforeEach(() => {
        wrapper = render(<RightSwipe callback={mockFn} /> )
    });

    it('should render correctly', () => {
        wrapper.getByTestId("rightSwipeBtn");
        wrapper.getByText("Delete");
    });

    it('should run callback on press', () => {
        fireEvent.press(wrapper.getByTestId("rightSwipeBtn"));
        expect(mockFn).toBeCalled()
    });
    
    it('should render correct styling', () => {
        
        expect(wrapper.getByTestId("rightSwipeBtn")).toHaveStyle({
            backgroundColor: 'red',
            justifyContent: 'center',
            alignItems: 'flex-end',
        });
        expect(wrapper.getByText("Delete")).toHaveStyle({
            color: "white",
        })
    });

    it('should match snapshot', () => {
        expect(wrapper.toJSON()).toMatchSnapshot();
    });
});