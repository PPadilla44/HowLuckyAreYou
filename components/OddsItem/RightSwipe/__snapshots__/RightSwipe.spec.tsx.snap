// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RightSwipe /> should match snapshot 1`] = `
<RNGestureHandlerButton
  collapsable={false}
  exclusive={true}
  onGestureEvent={[Function]}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  onHandlerStateChange={[Function]}
  rippleColor={0}
  style={
    Object {
      "alignItems": "flex-end",
      "backgroundColor": "red",
      "justifyContent": "center",
    }
  }
  testID="rightSwipeBtn"
>
  <View
    accessible={true}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontWeight": "bold",
          "paddingHorizontal": 30,
          "paddingVertical": 20,
        }
      }
    >
      Delete
    </Text>
  </View>
</RNGestureHandlerButton>
`;
